### ISSUE TITLE
su: Untracked pid for a terminal shell

### ISSUE THREAD

**Issue by seppzer0** [tiann/KernelSU#859]
**Describe the bug**
When trying to pop a Kali NetHunter terminal shell into chroot environment, it crashes with an error message `Untracked pid <pid>`.

**To Reproduce**
Steps to reproduce the behaviour:

_Note: Full reproduce is complicated, I'll specify only the key points for now and extend it later if required._
1. Build and flash Android kernel with NetHunter support;
2. Install KernelSU manager + NetHunter + NetHunter Terminal apps;
3. Grant SU permissions for both apps via KernelSU manager;
4. Open NetHunter Terminal app and launch permissions fix [script](https://github.com/seppzer0/s0nh_kernel/blob/dev/wrapper/modifications/nhpatch.sh), close the app;
5. Open NetHunter app, install the [chroot](https://kali.download/nethunter-images/current/rootfs/kalifs-arm64-minimal.tar.xz) (full/minimal, is not important);
6. Start the chroot, open NetHunter Terminal app, attempt to enter the chroot (select `Kali` option in the terminal app);
7. Observe the terminal app crash, extract logs via `dmesg`.

**Expected behaviour**
A shell (with the according pid) should be opened properly and receive SU permissions it requires.

**Screenshots**
![nhksu](https://github.com/tiann/KernelSU/assets/36642137/48f86989-5479-4cfa-8a83-073a6cd0a678)

**Smartphone (please complete the following information):**
 - Device: OnePlus 5T
 - OS: ParanoidAndroid
 - Version: Topaz 6 (-> A13)
 - Linux Kernel Version: 4.4.302

**Additional context**
Here is KernelSU [patch](https://github.com/seppzer0/s0nh_kernel/blob/dev/wrapper/modifications/dumplinger/ksu/4.4-compat.patch) I ended up with for now.
It fixed my previous problem with KernelSU not launching at all in the ROM.

**Comment by tiann** [comment]
I don't think this is a KernelSU issue.

**Comment by seppzer0** [comment]
There is no such issue if Magisk is used. Same kernel configuration (except KernelSU) is used.
That is why I assumed the problem might be in KernelSU.

I will investigate this more today. If you are sure this is not KernelSU-related, you can close the ticket.
Otherwise, I will close it myself when/if I find the root of the problem.

**Comment by tiann** [comment]
It works in Magisk doesn't mean it is a KernelSU issue, Magisk exists so long, it may have some special compatible code with Magisk.
Anyway, if you need our help. you should upload the bugreport in KernelSU settings.

**Comment by seppzer0** [comment]
> It works in Magisk doesn't mean it is a KernelSU issue, Magisk exists so long, it may have some special compatible code with Magisk.

Fair point.

> Anyway, if you need our help. you should upload the bugreport in KernelSU settings.

Of course, attaching the logs to this reply.
[KernelSU_bugreport_2023-08-12_10_46.tar.gz](https://github.com/tiann/KernelSU/files/12326876/KernelSU_bugreport_2023-08-12_10_46.tar.gz)

**Comment by seppzer0** [comment]
The name of the application that is crashing is `com.offsec.nhterm`.

**Comment by tiann** [comment]
Maybe you should try to disable `CONFIG_KPROBE`

**Comment by seppzer0** [comment]
Tried it now, didn't work.

But I started getting a new error:
`su: Failed to execute /usb/bin/zsh: Required key not available`
Which, based on some Google'ing, seems to be [specific](https://gitlab.com/kalilinux/nethunter/build-scripts/kali-nethunter-project/-/issues/1167) to Kali NetHunter itself.

I'm closing the issue as it indeed doesn't seem to be KernelSU-related.

Thank you for the suggestion, it provided a clue where to look next.
